summingbird-storm/src/main/scala/com/twitter/summingbird/storm/collector/AggregatorOutputCollector.scala

```diff
@@ -80,27 +90,36 @@ class AggregatorOutputCollector[K, V: Semigroup](
   Method wraps the adding the tuple to the spoutCache along with adding the corresponding
   messageId to the messageId Tracker.
    */
-  private def add(tuple: (K, V), streamid: String, messageId: Option[AnyRef] = None) = {
+  private def add(tuple: (K, V), streamid: String, messageId: Option[AnyRef] = None): Future[Map[K, V]] = {
     if (messageId.isDefined)
       trackMessageId(tuple, messageId.get, streamid)
     addToCache(tuple, streamid)
   }
 
-  private def addToCache(tuple: (K, V), streamid: String) = {
+  /*
+   * As there are separate summers for each stream. This method takes care of the stream lookup
+   * and adding the tuple to the corresponding cache.
+   */
+  private def addToCache(tuple: (K, V), streamid: String): Future[Map[K, V]] = {
     spoutCaches.get(streamid) match {
-      case Some(cac) => cac.add(tuple)
+      case Some(cache) => cache.add(tuple)
       case None => {
         spoutCaches(streamid) = summerBuilder.getSummer[K, V](implicitly[Semigroup[V]])
         spoutCaches(streamid).add(tuple)
       }
     }
   }
 
+  /**
+   * The lookup of stream is followed by the lookup on summerShard happens.
+   * The messageId is added to the corresponding group.
+   * All the messageIds are sent along with the crushed tuple.
+   */
   private def trackMessageId(tuple: (K, V), o: AnyRef, s: String): Unit = {
-    val messageIdTracker = streamMessageIdTracker.getOrElse(s, MMap[Int, MList[Object]]())
+    val messageIdTracker = cachesByStreamId.getOrElse(s, MMap[Int, MList[Object]]())
     var messageIds = messageIdTracker.getOrElse(summerShards.summerIdFor(tuple._1), MList())
```

Comment:
this does not need to be a var. You are calling the `+=` method on a mutable type, the reference itself never changes.